docs/query-building.md

```diff
@@ -93,3 +93,60 @@ var_dump($result->getResults());
 
 You can abstract parts of your query to separate classes and re-use them. More about it in the [re-useable fragments](re-useable-fragments.md)
 documentation.
+
+## Pagination
+
+You can use a pagination trait in your query like this:
+
+```php
+use ElasticSearcher\Abstracts\AbstractQuery;
+use ElasticSearcher\Fragments\Traits\PaginatedTrait;
+
+class MoviesYouMightLikeQuery extends AbstractQuery
+{
+  use PaginatedTrait;
+
+  public function setup()
+  {
+    $this->searchIn('suggestions', 'movies');
+
+    // Page 2 with 20 results per page.
+    $this->paginate(2, 20);
```

Comment:
You could technically do that since the method is public:

```php
$query = new MoviesYouMightLikeQuery($searcher);
$query->paginate(2,20);
$result = $query->run();
var_dump($result->getResults());
```

Working with Laravel this probably feels normal (you would even expect those last 2 lines to be omitted) tho.

I have some doubts about exposing those things (this would be a precedent) to the implementations of the query. My goals was to contain all query logic and building in the query class, so it (and all queries) can be consistently used (and can be predicted). If we start exposing too many things, a lot of logic would be done in the implementations.
 
Not sure, need some more insights or opinions about this.